File: src/main/java/com/catboyranch/roborancher/commands/CEdit.java
package com.catboyranch.roborancher.commands;

import com.catboyranch.roborancher.utils.ChannelUtils;
import com.catboyranch.roborancher.utils.CommandArgument;
import com.catboyranch.roborancher.utils.Utils;
import com.catboyranch.roborancher.configs.RoleType;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import com.catboyranch.roborancher.RoboRancher;
import com.catboyranch.roborancher.Server;

public class CEdit extends CommandBase {

    public CEdit(RoboRancher rancher) {
        super(rancher);
        name = "edit";
        description = "Edit messages sent by me!";
        allowedRoles = Utils.asArray(RoleType.ADMIN);
    }

    @Override
    public void run(String cmd, Server server, MessageReceivedEvent event, CommandResult result, CommandArgument... args) {
        if(args.length < 2) {
            result.error("Not enough arguments!");
            return;
        }

        String messageID = args[0].getText();
        String format = String.format("%s%s %s", server.getConfig().getCmdPrefix(), cmd, messageID);
        final String newMessage = event.getMessage().getContentRaw().replace(format, "");

        ChannelUtils.getMessage(server, messageID, objects -> {
            if(objects.length == 0) {
                result.error("Could not find message!");
                return;
            }
            ((Message)objects[0]).editMessage(newMessage).queue();
            result.success();
        });
    }

}

File: src/main/kotlin/com/catboyranch/roborancher/commands/CEdit.kt
package com.catboyranch.roborancher.commands

import com.catboyranch.roborancher.*
import net.dv8tion.jda.api.entities.Message
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class CEdit(rancher: RoboRancher): CommandBase(rancher) {
    override val name = "edit"
    override val description = "Edit messages sent by me!"
    override val allowedRoles = arrayOf(RoleUtils.RoleType.ADMIN)

    override fun run(cmd: String, server: Server, event: MessageReceivedEvent, result: CommandResult, vararg args: CommandArgument) {
        if(args.size < 2) {
            result.error("Not enough arguments!")
            return
        }
        val messageID = args[0].text
        val prefix = server.config.getData().cmdPrefix
        val newMessage = event.message.contentRaw.replace("$prefix$cmd $messageID", "")

        ChannelUtils.getMessage(server, messageID, object: IFunction {
            override fun run(vararg args: Any) {
                if(args.isEmpty()) {
                    result.error("Could not find message!")
                    return
                }
                val msg = args[0] as Message
                msg.editMessage(newMessage).queue()
                result.success()
            }
        })
    }
}